themes-default/slim/static/js/store.js

```diff
@@ -0,0 +1,195 @@
+    SOCKET_ONCLOSE,
+    SOCKET_ONERROR,
+    SOCKET_ONMESSAGE,
+    SOCKET_RECONNECT,
+    SOCKET_RECONNECT_ERROR,
+    ADD_CONFIG,
+    ADD_SHOW
+} = mutationTypes;
+
+const store = new Puex({
+    state: {
+        socket: {
+            isConnected: false,
+            message: '',
+            reconnectError: false
+        },
+        config: {
+            wikiUrl: null,
+            localUser: null,
+            posterSortdir: null,
+            locale: null,
+            themeName: null,
+            selectedRootIndex: null,
+            webRoot: null,
+            namingForceFolders: null,
+            cacheDir: null,
+            databaseVersion: null,
+            programDir: null,
+            animeSplitHomeInTabs: null,
+            layout: {
+                show: {
+                    allSeasons: null
+                }
+            },
+            dbPath: null,
+            nzb: null,
+            configFile: null,
+            fanartBackground: null,
+            trimZero: null,
+            animeSplitHome: null,
+            branch: null,
+            commitHash: null,
+            indexers: null,
+            sourceUrl: null,
+            rootDirs: null,
+            fanartBackgroundOpacity: null,
+            appArgs: null,
+            emby: {},
+            logDir: null,
+            sortArticle: null,
+            timePreset: null,
+            plex: {},
+            subtitles: {
+                enabled: null
+            },
+            fuzzyDating: null,
+            backlogOverview: null,
+            posterSortby: null,
+            kodi: {},
+            sslVersion: null,
+            pythonVersion: null,
+            comingEpsSort: null,
+            githubUrl: null,
+            datePreset: null,
+            subtitlesMulti: null,
+            os: null,
+            anonRedirect: null,
+            torrents: null
+        },
+        shows: [],
+        defaults: {
+            show: {
+                id: {
+                    tvdb: null,
+                    slug: null
+                },
+                rating: {
+                    imdb: {
+                        rating: null,
+                        votes: null
+                    }
+                },
+                country_codes: [], // eslint-disable-line camelcase
+                network: null,
+                airs: null,
+                config: {
+                    qualities: null,
+                    defaultEpisodeStatus: null,
+                    dvdOrder: null,
+                    seasonFolders: null,
+                    scene: null,
+                    sports: null,
+                    paused: null,
+                    location: null,
+                    airByDate: null,
+                    release: null,
+                    aliases: null,
+                    subtitlesEnabled: null,
+                    anime: null
+                }
+            }
+        }
+    },
+    mutations: {
+        [LOGIN_PENDING]() {},
```

Comment:
What sorcery is this? :scream: 
Can you explain why it's in brackets?

Reply:
const x = 'a';[x]

Reply:
const x = 'a';
const y = {};[a] =